src/otx/core/data/dataset/segmentation.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTXSegmentationDataset."""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import cv2
+import numpy as np
+import torch
+from datumaro.components.annotation import Image, Mask
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.segmentation import SegBatchDataEntity, SegDataEntity
+
+from .base import OTXDataset
+
+
+class OTXSegmentationDataset(OTXDataset[SegDataEntity]):
+    """OTXDataset class for segmentation task."""
+
+    def _get_item_impl(self, index: int) -> SegDataEntity | None:
+        item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
+
+        img = item.media_as(Image)
+        img_data = img.data
```

Comment:
Could you please using the method like below? We've already made the function to load the image data.
https://github.com/openvinotoolkit/training_extensions/blob/ee68ecd49ecd6780ba7e3915614a8ec5b216e499/src/otx/core/data/dataset/classification.py#L28C30-L28C30

Reply:
fixed [29dd2eb](https://github.com/openvinotoolkit/training_extensions/pull/2693/commits/29dd2eb8810801d217caef839453df331d47e086)